### PR TITLE
TST: stats: remove some unnecessary specification of dtype

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
gh-20625

#### What does this implement/fix?
Here are some other examples of simplifications that would result from gh-20625.

### CODE REVIEW

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -4830,7 +4830,7 @@ def pearsonr(x, y, *, alternative='two-sided', method=None, axis=0):
 
     dtype = xp.result_type(x.dtype, y.dtype)
     if xp.isdtype(dtype, "integral"):
-        dtype = xp.float64
+        dtype = xp.asarray(1.).dtype
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:4833]: This is actually orthogonal related to default dtypes, but it has come up a few times: what should we do with integer arrays - reject them, convert to a fixed dtype like `float64`, or convert them to the default dtype of the given backend?

**Review by j-bowhay — APPROVED** [review]
Thanks!

File: scipy/stats/_stats_py.py (hunk 1 above)
Comment (j-bowhay) [reply scipy/stats/_stats_py.py:4833]: Hmm good question, maybe a discussion point for https://github.com/scipy/scipy/issues/18286; I am not sure I have thought about it enough yet to form an opinion
